### ISSUE TITLE
Next Phase and Contest deadlines are incorrect on challenge details

### ISSUE THREAD

**Issue by SathyaJayabal** [topcoder-platform/community-app#4723]
Next Phase and Contest deadlines are incorrect on challenge details
examples
https://beta-community-app.topcoder.com/challenges/934bb8ac-1269-469b-a1b8-64b6443a6f9f
https://www.topcoder.com/challenges/30136288
<img width="1440" alt="Screenshot 2020-08-07 at 1 57 48 PM" src="https://user-images.githubusercontent.com/58783823/89626030-66e2ab80-d8b6-11ea-8ed4-1fdce20f208b.png">

**Comment by SathyaJayabal** [comment]
@sushilshinde , api response seems to be correct
<img width="1440" alt="Screenshot 2020-08-07 at 2 04 24 PM" src="https://user-images.githubusercontent.com/58783823/89626398-fb4d0e00-d8b6-11ea-8174-1eeb1275dd92.png">

**Comment by crazyk07** [comment]
Contest https://www.topcoder.com/challenges/30136485 has been created for this ticket.<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by crazyk07** [comment]
Contest https://www.topcoder.com/challenges/30136485 has been updated - it has been assigned to nursoltan-s.<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by nursoltan-s** [comment]
![Screenshot from 2020-08-10 23-28-10](https://user-images.githubusercontent.com/7783676/89800486-75a0bb00-db61-11ea-8cb3-0c80c420d33f.png)

**Comment by nursoltan-s** [comment]
@sushilshinde 
requested PR here: https://github.com/topcoder-platform/community-app/pull/4732

**Comment by sushilshinde** [comment]
Thanks @nursoltan-s will review and merge

**Comment by SathyaJayabal** [comment]
issue not fixed
https://beta-community-app.topcoder.com/challenges/934bb8ac-1269-469b-a1b8-64b6443a6f9f
https://www.topcoder.com/challenges/30136288
<img width="1440" alt="Screenshot 2020-08-11 at 8 33 13 AM" src="https://user-images.githubusercontent.com/58783823/89852835-60517e00-dbad-11ea-9596-0a0aa6e4ff55.png">

**Comment by SathyaJayabal** [comment]
@sushilshinde @nursoltan-s , when there are no more phases after the review phase (appeals/appeals response etc) we do not display the review phase. Reducing the priority and removing the Must fix label, as the main issue(correct dates) is fixed.

**Comment by nursoltan-s** [comment]
https://github.com/topcoder-platform/community-app/pull/4735
Fixed above PR.

**Comment by SathyaJayabal** [comment]
@nursoltan-s , we should not remove the review phase when we have other phases after that For example:
https://beta-community-app.topcoder.com/challenges/64e290dd-8a33-4132-a30b-7c260d7793ba
https://www.topcoder.com/challenges/30135712?tab=details

<img width="1440" alt="Screenshot 2020-08-11 at 11 23 20 AM" src="https://user-images.githubusercontent.com/58783823/89862413-897e0880-dbc5-11ea-896e-0b19095f7b67.png">

**Comment by nursoltan-s** [comment]
@sushilshinde @SathyaJayabal 
For this case, there is a phase after "Review", which is "Approval". We should add "Review phase", right? But this doesn't match with old page.
Which one is correct?

![Screenshot from 2020-08-11 21-32-00](https://user-images.githubusercontent.com/7783676/89903586-6a5d9600-dc1a-11ea-97b3-bf99add7d701.png)

**Comment by SathyaJayabal** [comment]
@nursoltan-s , we can go with existing page.

**Comment by nursoltan-s** [comment]
@SathyaJayabal you mean "topcoder-dev.com" page?

**Comment by SathyaJayabal** [comment]
@nursoltan-s, yes. You can also refer to production/beta for examples of V4/V5 challenges, as production has better data.
https://beta-community-app.topcoder.com/challenges?
https://www.topcoder.com/challenges

So the logic must be the same as V4.

**Comment by nursoltan-s** [comment]
PR: https://github.com/topcoder-platform/community-app/pull/4740

**Comment by SathyaJayabal** [comment]
@nursoltan-s , The above issue is not fixed
https://www.topcoder.com/challenges/30135712?tab=details
https://staging-community-app.topcoder.com/challenges/64e290dd-8a33-4132-a30b-7c260d7793ba

<img width="1440" alt="Screenshot 2020-08-13 at 11 52 48 AM" src="https://user-images.githubusercontent.com/58783823/90101128-ba904180-dd5b-11ea-918c-914591694d81.png">

Also in the case of F2F challenges, there are some variations 
example:
https://staging-community-app.topcoder.com/challenges/4650939b-2f51-471c-bef7-bb1a7d7dc381
https://www.topcoder.com/challenges/30136921

<img width="1440" alt="Screenshot 2020-08-13 at 11 54 59 AM" src="https://user-images.githubusercontent.com/58783823/90101221-ef9c9400-dd5b-11ea-9653-d1e6f2af06c0.png">

cc @sushilshinde

**Comment by SathyaJayabal** [comment]
here is another example where next  phase is incorrect
https://www.topcoder.com/challenges/30135512
https://staging-community-app.topcoder.com/challenges/68f0ebff-314d-4c3c-87ac-3787b6a53783
<img width="1440" alt="Screenshot 2020-08-13 at 1 10 52 PM" src="https://user-images.githubusercontent.com/58783823/90107513-8e2df280-dd66-11ea-8504-61f8faf62f46.png">

**Comment by nursoltan-s** [comment]
PR: https://github.com/topcoder-platform/community-app/pull/4777

**Comment by sushilshinde** [comment]
<img width="1792" alt="prod-v5-zurich" src="https://user-images.githubusercontent.com/683231/91146647-ca514380-e6d4-11ea-9584-921dd6552d71.png">

**Comment by sushilshinde** [comment]
@nursoltan-s please check 
https://staging-community-app.topcoder.com/challenges/7f7762e5-fe0c-4a9a-a361-58e60780df14

![image](https://user-images.githubusercontent.com/683231/91146791-fd93d280-e6d4-11ea-8d88-56e551224571.png)

**Comment by sushilshinde** [comment]
I'm reverting the PR

**Comment by nursoltan-s** [comment]
@sushilshinde 
I created new PR against "integration-v5-challenge-api" branch. Can you check that again? Thanks
https://github.com/topcoder-platform/community-app/pull/4812

**Comment by sushilshinde** [comment]
@nursoltan-s it's breaking again

![CF Dev F2F and Task 2020-08-31 11-50-22](https://user-images.githubusercontent.com/683231/91688687-2eb24e00-eb80-11ea-80b7-9d0bd0ebd312.png)

I'm reverting again.

**Comment by crazyk07** [comment]
Contest https://www.topcoder.com/challenges/30136485 has been updated - it has been assigned to cagdas001.<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by luizrrodrigues** [comment]
@cagdas001 Any progress in this one?

**Comment by luizrrodrigues** [comment]
@cagdas001 Any status? We need closed this one ASAP.

Thanks

**Comment by cagdas001** [comment]
PR: #4842

**Comment by SathyaJayabal** [comment]
verified on staging
<img width="1440" alt="Screenshot 2020-09-02 at 12 10 15 PM" src="https://user-images.githubusercontent.com/58783823/91942582-ce0a4900-ed18-11ea-9e81-6af33156f9fe.png">
<img width="1440" alt="Screenshot 2020-09-02 at 12 14 55 PM" src="https://user-images.githubusercontent.com/58783823/91942589-d2cefd00-ed18-11ea-8d76-4b37ac8d2fd0.png">
<img width="1440" alt="Screenshot 2020-09-02 at 12 17 37 PM" src="https://user-images.githubusercontent.com/58783823/91942593-d4002a00-ed18-11ea-93c2-5b2d3cd5cd08.png">

**Comment by SathyaJayabal** [comment]
verified on staging (develop branch)
<img width="1440" alt="Screenshot 2020-09-02 at 3 43 10 PM" src="https://user-images.githubusercontent.com/58783823/91969945-8abdd380-ed34-11ea-981a-2d246887f273.png">
<img width="1440" alt="Screenshot 2020-09-02 at 3 45 39 PM" src="https://user-images.githubusercontent.com/58783823/91969957-91e4e180-ed34-11ea-9914-2a551a25d109.png">
<img width="1440" alt="Screenshot 2020-09-02 at 3 53 35 PM" src="https://user-images.githubusercontent.com/58783823/91969965-94473b80-ed34-11ea-95ff-746c9a700b48.png">

**Comment by SathyaJayabal** [comment]
verified on production
<img width="1440" alt="Screenshot 2020-09-03 at 9 36 01 AM" src="https://user-images.githubusercontent.com/58783823/92071133-8e506980-edcb-11ea-87e9-e5bebfd8e000.png">

<img width="1440" alt="Screenshot 2020-09-03 at 9 35 59 AM" src="https://user-images.githubusercontent.com/58783823/92071118-88f31f00-edcb-11ea-8d27-1e271d3faf2a.png">

**Comment by crazyk07** [comment]
Payment task has been updated: https://software.topcoder.com/review/actions/ViewProjectDetails?pid=30136485<br/><br/>```This is an automated message for crazyk via Topcoder X```
